src/gt4py/next/common.py

```diff
@@ -574,22 +574,7 @@ def __call__(self, func: fbuiltins.BuiltInFunction[_R, _P], /) -> Callable[_P, _
             ...
 
 
-# TODO(havogt): replace this protocol with the new `GTFieldInterface` protocol
-class NextGTDimsInterface(Protocol):
-    """
-    Protocol for objects providing the `__gt_dims__` property, naming :class:`Field` dimensions.
-
-    The dimension names are objects of type :class:`Dimension`, in contrast to
-    :mod:`gt4py.cartesian`, where the labels are `str` s with implied semantics,
-    see :class:`~gt4py._core.definitions.GTDimsInterface` .
-    """
-
-    @property
-    def __gt_dims__(self) -> tuple[Dimension, ...]:
-        ...
-
-
-# TODO(egparedes): add support for this new protocol in the cartesian module
+# TODO(havogt): we need to specify when we should use this interface vs the `Field` protocol.
```

Comment:
I assume this Protocol should be used if users define their own buffers that should be treated as fields. `__gt_domain__` is not enough, we need to specify how to get the buffer.
- Additionally: do want models of the concept to be also directly usable as field? or should this interface only be used to extract a buffer (that we could then wrap (possibly with host<->device copies) into a proper Field for embedded). Probably the latter makes sense, as the former is the `Field` interface.
- Note: forcing to add a member to a user object that returns an instance of a gt4py class, doesn't feel nice. Maybe we should at least allow `-> DomainLike` for `__gt_domain__`. Additionally, we would need `Dimension` to be any hashable object (similar to xarray) to be compatible in that sense.
- We should implement the same utility function for `__gt_domain__` that we have for `__gt_dims__`: not access the property directly but via the function, then we can provide default implementation for known types or additionally provide a single dispatch point for extending the list of default implementations.